src/OpenSage.Tools.BigEditor/UI/FileBrowser.cs

```diff
@@ -0,0 +1,335 @@
+                {
+                    if (ImGui.IsMouseDoubleClicked(0))
+                    {
+                        _currentPath = adapter.RootDirectory;
+                    }
+                }
+
+                ImGui.NextColumn();
+                ImGui.NextColumn();
+                ImGui.NextColumn();
+            }
+
+            int i = 0;
+            foreach (var entry in adapter.Entries)
+            {
+                if (ImGui.Selectable(entry.Name, i == _currentSelection, ImGuiSelectableFlags.SpanAllColumns | ImGuiSelectableFlags.AllowDoubleClick))
+                {
+                    if (ImGui.IsMouseDoubleClicked(0))
+                    {
+                        if (entry.IsFile)
+                        {
+                            ImGui.CloseCurrentPopup();
+
+                            return true;
+                        }
+                        else
+                        {
+                            i = -1;
+                            _currentPath = entry.FullName;
+                        }
+                    }
+
+                    _currentSelection = i;
+                    _currentFile = entry.FullName;
+                }
+
+                ImGui.NextColumn();
+
+                ImGui.Text(entry.CreationTime.ToLocalTime().ToString());
+
+                ImGui.NextColumn();
+
+                ImGui.Text(ImGuiUtility.GetFormatedSize(entry.Length));
+
+                ImGui.NextColumn();
+
+                i++;
+            }
+
+            ImGui.EndChild(); // browser
+
+            ImGui.EndChild(); // container
+
+            return false;
+        }
+
+        private static bool DirTree(string currentPath = "/")
+        {
+            DirectoryInfo current = new DirectoryInfo(currentPath);
+            DirectoryInfo[] dirs = current.GetDirectories();
+
+            if (ImGui.TreeNodeEx(currentPath, ImGuiTreeNodeFlags.DefaultOpen))
+            {
+                foreach (var dir in dirs)
+                {
+                    SubTree(dir.Name + Path.DirectorySeparatorChar, Path.DirectorySeparatorChar + dir.Name + Path.DirectorySeparatorChar);
+                }
+
+                ImGui.TreePop();
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool SubTree(string label, string path)
+        {
+            DirectoryInfo current = null;
+            DirectoryInfo[] dirs = null;
+
+            try {
+                current = new DirectoryInfo(path);
+                dirs = current.GetDirectories();
+            }
+            catch (Exception e) {
+                if (ImGui.TreeNodeEx(label)) {
+                    ImGui.Text(e.Message);
+                    ImGui.TreePop();
+                }
+                return false;
+            }
+
+            var currentSelect = Regex.Match($"{_currentPath}/", $"{path}");
```

Comment:
Crashes for me:
```
System.ArgumentException: "parsing '\$RECYCLE.BIN\' - Illegal \\ at end of pattern."
```

Reply:
Maybe the problem in regex, because in this case regex contain not shielded "$" char

Reply:
Or not. I created $RECYCLE.BIN directory and work fine.
Ok, I'll try to test this on Windows

Reply:
Fixed